athena/utils.py

```diff
@@ -177,3 +178,116 @@ def sort_eigpairs(matrix):
     s[s == 0] = 1
     evects *= s
     return evals.reshape(-1, 1), evects
+
+
+class CrossValidation():
+    """doc"""
+    def __init__(self,
+                 inputs=None,
+                 outputs=None,
+                 gradients=None,
+                 folds=5,
+                 subspace=None,
+                 gp_dimension=1,
+                 **kwargs):
+
+        self.inputs = inputs
+        self.outputs = outputs
+        self.gradients = gradients
+        self.folds = folds
+        self.ss = subspace
+        self.gp_dimension = gp_dimension
+        self.gp = None
+        self.kwargs = kwargs
+
+    def cross_validation(self):
+        """doc"""
+        mask = np.arange(self.inputs.shape[0])
+
+        np.random.shuffle(mask)
+        scores = np.zeros((self.folds))
+        s_mask = np.array_split(mask, self.folds)
+
+        for i in range(self.folds):
+            v_mask = s_mask[i]
+            validation = (self.inputs[v_mask, :], self.outputs[v_mask, :])
+
+            t_mask = ~v_mask
+            self.fit(self.inputs[t_mask, :], self.gradients[t_mask, :, :],
+                     self.outputs[t_mask, :])
+
+            scores[i] = self.scorer(validation[0], validation[1])
+
+        return scores.mean(), scores.std()
+
+    def training(self):
```

Comment:
where do you use this method? I would call it `train`. Maybe you can transform it in a private method and use it inside the `cross_validation` method with the proper args